evmrpc/filter.go

```diff
@@ -46,14 +69,304 @@ type filter struct {
 	lastToHeight int64
 }
 
+// Global worker pool
+type WorkerPool struct {
+	workers   int
+	taskQueue chan func()
+	once      sync.Once
+	done      chan struct{}
+	wg        sync.WaitGroup
+}
+
+var (
+	globalWorkerPool *WorkerPool
+	poolOnce         sync.Once
+	// Semaphore to limit concurrent I/O operations against the database
+	dbReadSemaphore  = make(chan struct{}, MaxDBReadConcurrency)
+	globalBlockCache = NewBlockCache(1000)
+
+	// every request consumes 1 token, regardless of block range
+	globalRPSLimiter = rate.NewLimiter(rate.Limit(GlobalRPSLimit), GlobalRPSLimit)
+	// Global bloom cache - smaller than block cache since blooms are tiny
+	globalBloomCache = NewBloomCache(50000)
```

Comment:
Bloom and receipts are belong to a single block any way, can we just combine them into the blockcache itself?